src/src/Controller/AbandonedCartController.php

```diff
@@ -0,0 +1,254 @@
+            }
+
+            if (!$this->send($cart)) {
+                ++$failCount;
+                $success = false;
+            }
+
+            ++$count;
+        }
+
+        echo sprintf("%s abandoned cart email(s) sent!\r\n", $count);
+        if (!$success) {
+            echo sprintf("%s failed cart(s), check logs!\r\n", $failCount);
+        }
+
+        return $success;
+    }
+
+    private function isDelayTimePassed(AbandonedCart $cart): bool
+    {
+        $cart_updated_time = strtotime($cart->dateUpdated);
+        $syncInterval = $this->configuration->get('SMAILY_ABANDONED_CART_TIME');
+
+        $reminder_time = strtotime('+' . $syncInterval . ' minutes', $cart_updated_time);
+        $current_time = strtotime(date('Y-m-d H:i') . ':00');
+        // Don't continue if cart delay time has not passed.
+        if ($current_time < $reminder_time) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Send abandoned cart emails to customers.
+     *
+     * @return void
+     */
+    private function send(AbandonedCart $cart): bool
+    {
+        $response = $this->api->triggerAutomation(
+            $this->configuration->get('SMAILY_CART_AUTORESPONDER'),
+            [$this->generatePayload($cart)],
+        );
+
+        if ($response->getStatusCode() !== 200) {
+            Logger::logErrorWithFormatting(
+                'Failed sending out abandoned cart email for email: %s, cart_id: %s. ' .
+                'Smaily response HTTP response code: %s.',
+                $cart->email,
+                $cart->cartID,
+                $response->getStatusCode()
+            );
+
+            return false;
+        }
+
+        $body = json_decode($response->getBody()->getContents(), true);
+        if (!isset($body['code']) || $body['code'] !== 101) {
+            Logger::logErrorWithFormatting(
+                'Failed sending out abandoned cart email for email: %s, cart_id: %s. ' .
+                'Smaily response code: %s, message: %s.',
+                $cart->email,
+                $cart->cartID,
+                $body['code'],
+                $body['message']
+            );
+
+            return false;
+        }
+
+        $sentCart = new SentCart();
+        $sentCart->id_customer = $cart->customerID;
+        $sentCart->id_cart = $cart->cartID;
+        $sentCart->date_sent = date('Y-m-d H:i:s');
+        $sentCart->save();
+
+        return true;
+    }
+
+    private function generatePayload(AbandonedCart $cart): array
+    {
+        $payload = [
+            'email' => $cart->email,
+        ];
+
+        $syncAdditional = json_decode($this->configuration->get('SMAILY_CART_SYNCRONIZE_ADDITIONAL'), true);
+        if (in_array('first_name', $syncAdditional) && $syncAdditional['first_name'] === true) {
+            $payload['first_name'] = $cart->firstName;
+        }
+
+        if (in_array('last_name', $syncAdditional) && $syncAdditional['last_name'] === true) {
```

Comment:
isset instead of in_array? otherwise in that case: $syncAdditional = ['not_last_name' = last_name']; we will get undefined index here